File: CPP/Lee1447.cpp
//
// Created by yangf on 2022/2/11.
//
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

class Solution {
public:
    vector<string> simplifiedFractions(int n) {
        vector<string> result;
        for (int i = 2; i <= n; i++) {
            for (int j = 1; j < i; j++) {
                if (__gcd(i, j) == 1) {
                    result.push_back(to_string(j) + "/" + to_string(i));
                }
            }
        }
        return result;
    }
};

File: JAVA/Lee1447.java
package JAVA;

import java.util.ArrayList;
import java.util.List;

public class Lee1447 {
    public List<String> simplifiedFractions(int n) {
        List<String> result = new ArrayList<>();
        for (int i = 2; i <= n; i++) {
            for (int j = 1; j < i; j++) {
                if (gcd(i, j) == 1) {
                    result.add(i + "/" + j);
                }
            }
        }
        return result;
    }

    public int gcd(int a, int b) {
        return b != 0 ? gcd(b, a % b) : a;
    }
}
